packages/gestalt/src/PopoverEducational.js

```diff
@@ -12,50 +13,51 @@ import { type Indexable } from './zIndex.js';
 type Size = 'sm' | 'flexible';
 type IdealDirection = 'up' | 'right' | 'down' | 'left';
 type Role = 'dialog' | 'tooltip';
-type PrimaryActionType = {|
-  accessibilityLabel?: string,
-  href?: string,
-  text: string,
-  onClick?: ({|
-    event:
-      | SyntheticMouseEvent<HTMLButtonElement>
-      | SyntheticMouseEvent<HTMLAnchorElement>
-      | SyntheticKeyboardEvent<HTMLAnchorElement>
-      | SyntheticKeyboardEvent<HTMLButtonElement>,
-    dangerouslyDisableOnNavigation: () => void,
-  |}) => void,
-  rel?: 'none' | 'nofollow',
-  target?: null | 'self' | 'blank',
-|};
+type PrimaryActionType =
+  | {|
+      accessibilityLabel?: string,
+      href: string,
+      onClick?: ({|
+        event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+        dangerouslyDisableOnNavigation: () => void,
+      |}) => void,
+      rel?: 'none' | 'nofollow',
+      role: 'link',
+      target?: null | 'self' | 'blank',
+      text: string,
+    |}
+  | {|
+      accessibilityLabel?: string,
+      onClick?: ({|
+        event: SyntheticMouseEvent<HTMLButtonElement> | SyntheticKeyboardEvent<HTMLButtonElement>,
+      |}) => void,
+      role?: 'button',
+      text: string,
+    |};
 
-function PrimaryAction({
-  accessibilityLabel,
-  href,
-  text,
-  onClick,
-  rel,
-  target,
-}: PrimaryActionType) {
-  return href ? (
-    <Button
-      accessibilityLabel={accessibilityLabel}
-      color="white"
-      fullWidth={false}
-      href={href}
-      onClick={onClick}
-      rel={rel}
-      role="link"
-      target={target}
-      text={text}
-    />
-  ) : (
+function PrimaryAction(props: PrimaryActionType) {
+  if (props.role === 'link') {
+    return (
+      <ButtonLink
+        accessibilityLabel={props.accessibilityLabel}
+        color="white"
+        fullWidth={false}
+        href={props.href}
+        onClick={props.onClick}
+        rel={props.rel}
+        target={props.target}
+        text={props.text}
+      />
+    );
+  }
+  return (
     <Button
-      accessibilityLabel={accessibilityLabel}
+      accessibilityLabel={props.accessibilityLabel}
       color="white"
       fullWidth={false}
-      onClick={onClick}
-      role="button"
-      text={text}
+      onClick={({ event }) => props.onClick?.({ event })}
+      type="button"
```

Comment:
do we need `type` here?

Reply:
Hmm don't think so, just left it there as before. Removed.